### ISSUE TITLE
Change Screen Size

### ISSUE THREAD

**Issue by ve3sjk** [AndreiDegtiarev/AWind#7]
I am porting the app to use a 7" Display. I have some of the examples up and running. My display is 800 x 600 i am wondering if you might point me in the right direction in the files for changing the screen size from the original. I see many references to width and height in some of the files but don't see a place where the size is read from the display or set as a variable constant. For example i can run the sensor monitor example but the sensors are squished up to the top left quadrant of my display while the day night button seems to be perfectly aligned with the bottom right corner of the display making me thing at some point it knows my display is 800 x 600 or it would not have been able to place the day night button where it is. Any insight would be greatly appreciated. Another example in the sensor monitor there is a red line around the outside border in the display i need to move that border in a little but i don't know which file is actually drawing that border on the main window.

**Comment by ve3sjk** [comment]
Pretty sure i have found most places to start changing the screen size but i found an issue with the alarm display in the sensor monitor. When the sensor is in alarm the box turns red but there are two lines to the left of it. The lines don't disappear when the box goes back to normal color. I have found the ref to where the decorator alarm color is set but i am unsure where in the code the extra lines are being generated. I have attached a picture of the issue. Issue follows any sensor that is in alarm display.

![20170910_223527s](https://user-images.githubusercontent.com/7632604/30258984-e98bfb74-968b-11e7-828d-0dc61a3b4d67.jpg)

**Comment by elpapais** [comment]
Hi.

Work fine also on 9" dispaly.
Actually is near the limit of Arduino Due but i'm using seven dialog window and 59Buttons, 32 Buttons window, 72 labels, 4 radio, 18 Text numbers, 8 round gauge, 4 timers, 4 radial gauge.
Is much interessant if come out a image button control.
regards

**Comment by ve3sjk** [comment]
Are you seeing the lines in the image i posted when in alarm. I cannot find where in the software those lines are being created can't really start writing my application until i fix those lines

**Comment by elpapais** [comment]
i have the same problem in the example  Sensors.
I think is a bug for character or objects one over other.
You made same as the sample in other system.
the library is verry extra!!!
I Hope Andrei will complete whit a image-button or image container and wil be verry complete.

**Comment by ve3sjk** [comment]
Starting to get a handle on how all the files interact with each other I am assuming that those lines have something to do with the section that handles decorators eventually i will find the instance. I am leaning towards the drawrect function where the lines are supposed to be a box. What kind of image would it be a font, bit map stored in progmem? I am coding for mega2560 since my application needs 5v logic on the outputs and the due is 3.3. I have SSD1963 TFT which has 64mb of flash on the tft that could be utilized to store the images. Might research how its done in other libraries and port something over.

**Comment by AndreiDegtiarev** [comment]
Hi, I'm happy to see that you solve all problems by our own, except probably image-button :-) 
For the start I would advise to use VacuumPump example. Because the architecture there is cleaner as in SensorsMonitor and I expect you will have less problems with red lines. 
@ve3sjk. Do you have any plans to use AWind library for specific application?
Andrei

**Comment by ve3sjk** [comment]
I never did find where the red lines are coming from also noticed a problem where the left box line for graphs is shifted way over to the left when you click for the graph. Picture below lol showing for that i need to clean my screen lol. I will switch to the vacuumPump example and see how it goes. I am working on using this as the menu for controlling by Bradley smoker via a PID loop for SSR with profiles and sd card logging basically process control.

lines appear when moving between sec and minute chart options. On first two sensor windows seem to be affected.

![20170913_203440s](https://user-images.githubusercontent.com/7632604/30407018-b707282e-98c3-11e7-8c84-1804ec635521.jpg)

**Comment by ve3sjk** [comment]
I am using a download zip from each of the three libraries from github for awind, asensor, aframe. I get the following error when i try and compile the vacuumpump example.

![error](https://user-images.githubusercontent.com/7632604/30407838-d927c22e-98c8-11e7-9dfc-6c2410ac9b5a.png)

**Comment by elpapais** [comment]
The two white line in the middle of the screen are from the box while you start to 0. Try to start draving not from 0,0 but 1,1 and maybe don't  appear animore.

**Comment by AndreiDegtiarev** [comment]
I agree with elpapais that those lines are probably kind of low level artefacts, and not "wrong rectangles" in application code. 
The MAXFLOAT is system variable that is defined in math.h if you compile project for Due and it isn't part of AWind library.
For "Bradley smoker" the VacuumPump is very good example. Because there are kind of control algorithms implemented and the architecture is clearly split between visualization and hardware code.
It is funny but for my actual project I also have smoker as pilot application. :-)

**Comment by ve3sjk** [comment]
Does the vacuumpump example only work on the due. I am compiling for mega2560 since my app needs 5v outputs, really don't want to wire the level convertors unless i need to. Can't seems to get around the maxfloat error even if i include math.h. I have owned a few bradleys actually go caught into the hype of the new smart smoker with blue tooth, a great disappointment lol. Basically a blue tooth app that does not work and the same old bradley with a 25 degree swing in temp. I am doing like the others and adding the second heating element. I have used industrial PID but what i really want is heating profiles which are more user friendly than a ramp and soak industrial PID and of course logging.

**Comment by AndreiDegtiarev** [comment]
It should work on Mega. You can try to define MAXFLOAT macro in your code. In your project because of safety reasons you can also think about splitting of control logic and visualization between different boards. The project itself is a cool one and I hope at the end you will report with some nice pictures also from "end result" :-)
